src/differentiable_vectors/autodiff.jl

```diff
@@ -13,7 +13,7 @@ function OnceDifferentiable(f!, F::AbstractArray, x::AbstractArray, autodiff = :
             F = similar(x)
             fj!(F, J, x)
         end
-        return OnceDifferentiable(f!, j!, fj!, similar(x), similar(x))
+        return OnceDifferentiable(f!, j!, fj!, x, x)
```

Comment:
is it fine that this aliases?

Reply:
yes https://github.com/JuliaNLSolvers/NLSolversBase.jl/pull/27/files#diff-823994a5d1ac00d5d4a924d9a38dcbeeR26